homework-submission/week3/src/App.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+import './App.css';
+import Description from './ToDo/Description';
+import DeadLines from './ToDo/DeadLine';
+import DoneCheckBox from './ToDo/DoneCheckBox';
+import todos from './todoList';
+/** WEEK 3 */
+import NewToDoComponent from './ToDo/NewToDoComponent';
+import ControlsComponent from './ToDo/ControlsComponent';
+
+let oldDescription = '';
+class App extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos,
+      editMode: -1
+    };
+
+  }
+
+  handleCheck = (x) => {
+    let tempTodos = [...this.state.todos];
+    tempTodos[x].done = !tempTodos[x].done;
+    this.setState({
+      todos: tempTodos
+    })
+  }
+
+
+  handleSubmit = (newTodo) => {
+    this.setState((state) => ({
+      todos: [...state.todos, newTodo]
+
+    }));
+
+  }
+
+
+  handleRemoveToDo = (indexToDo) => {
+    const tempTodos = [...this.state.todos];
+    tempTodos.splice(indexToDo, 1);
+
+    this.setState((state) => ({
+      todos: tempTodos
+    }))
+
+  }
+
+  handleEdit = (indexToDo) => {
+    this.setState({ editMode: indexToDo });
+  }
+
+  handleUpdate = (updateToDo, indexToDo) => {
+    const tempToDos = [...this.state.todos];
+    tempToDos[indexToDo].description = updateToDo;
+    oldDescription = '';
+
+    this.setState({
+      todos: tempToDos,
+      editMode: -1
+    });
+  }
+  handleCancel = (description, index) => {
+    const tempToDos = [...this.state.todos];
+    tempToDos[index].description = oldDescription;
+    oldDescription = '';
+
+    this.setState(
+      {
+        todos: tempToDos,
+        editMode: -1
+      }
+    )
+    this.setState({ editMode: -1 });
+  }
```

Comment:
you don't need the code from line 66 to 75 
because in CANCEL you don't need to change anything except the editMode, so the code in line 76 is enough.